lte/gateway/python/magma/mobilityd/rpc_servicer.py

```diff
@@ -233,13 +233,16 @@ def AllocateIPAddress(self, request, context):
                 composite_sid + ",ipv6", IPAddress.IPV6,
                 context, request,
             )
-            ipv4_addr = ipv4_response.ip_list[0]
-            ipv6_addr = ipv6_response.ip_list[0]
-            # Get vlan from IPv4 Allocate response
-            resp = AllocateIPAddressResponse(
-                ip_list=[ipv4_addr, ipv6_addr],
-                vlan=ipv4_response.vlan,
-            )
+            ip_addresses = list(ipv4_response.ip_list)[:1] + list(ipv6_response.ip_list)[:1]
+            if len(ip_addresses) < 2:
+                logging.warning("IPv4/IPv6 IP address allocation not successful")
+            if ip_addresses:
+                resp = AllocateIPAddressResponse(
```

Comment:
@ardzoht I tested this locally with a Python grpc client, and this doesn't work if one of the two allocations fails. The reason is that we delegate the allocation of the IP addresses to the `_get_allocate_ip_response` function, and that function sets the statuscode for the response, making the client code raise an exception. It looks to me like the `_get_allocate_ip_response` function shouldn't be called twice within one grpc request. However, I think changing that is out of scope of this issue, as it's mostly trying to get rid of the `IndexError`. I suggest I revert this to my initial version that returns both addresses or none, and I create a separate issue for handling the case of only one successfully allocated address.

Reply:
@jheidbrink Makes sense, thanks for double verifying the issue. If we can open a separate issue for the one-only allocation that sounds good by me and I will accept the PR with the previous implementation.

Reply:
Sounds good! I force-pushed the old version and will create an issue for the one-only allocation tomorrow.